aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -991,6 +992,15 @@ void apply_lu_solve(const Tensor& LU, const Tensor& pivots, const Tensor& B, Tra
 
 // This is a type dispatching helper function for 'apply_lu_solve'
 void lu_solve_kernel(const Tensor& LU, const Tensor& pivots, const Tensor& B, TransposeType trans) {
+  // Lapack will write into unrelated memory if pivots are not in the right range so we do
+  // some simple sanity checks here.
```

Comment:
Mention that we just run this on CPU because syncs and blablabla

Reply:
Why do we do these checks here? The ones in the kernel are not sufficient?